cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiff.java

```diff
@@ -0,0 +1,203 @@
+            + "|.*-909[1-4]\\.ssl\\.truststore\\.password"
+            + "|.*-909[1-4]\\.ssl\\.truststore\\.type"
+            + "|.*-909[1-4]\\.ssl\\.client\\.auth"
+            + "|.*-909[1-4]\\.scram-sha-512\\.sasl\\.jaas\\.config"
+            + "|.*-909[1-4]\\.sasl\\.enabled\\.mechanisms"
+            + "|advertised\\.listeners"
+            + "|zookeeper\\.connect"
+            //+ "|log\\.dirs"*/
+            + "|super\\.users"
+            + "|broker\\.rack)$");
+
+    public KafkaBrokerConfigurationDiff(Map<ConfigResource, Config> current, ConfigMap desired, KafkaVersion kafkaVersion, int brokerId) {
+        this.current = current;
+        this.diff = emptyKafkaConf(); // init
+        this.desired = desired;
+        this.kafkaVersion = kafkaVersion;
+        this.brokerId = brokerId;
+        Config brokerConfigs = this.current.get(new ConfigResource(ConfigResource.Type.BROKER, Integer.toString(brokerId)));
+        if (brokerConfigs == null) {
+            log.warn("Failed to get broker {} configuration", brokerId);
+        } else {
+            this.currentEntries = brokerConfigs.entries();
+            this.diff = computeDiff();
+        }
+    }
+
+    private KafkaConfiguration emptyKafkaConf() {
+        Map<String, Object> conf = new HashMap<>();
+        return new KafkaConfiguration(conf.entrySet());
+    }
+
+
+    private void fillPlaceholderValue(Map<String, String> orderedProperties, String placeholder, String value) {
+        orderedProperties.entrySet().forEach(entry -> {
+            entry.setValue(entry.getValue().replaceAll("\\$\\{" + Pattern.quote(placeholder) + "\\}", value));
+        });
+    }
+
+    public KafkaConfiguration getDiff() {
+        return this.diff;
+    }
+
+    public boolean isDesiredPropertyDefaultValue(String key) {
+        Optional<ConfigEntry> entry = currentEntries.stream().filter(configEntry -> configEntry.name().equals(key)).findFirst();
+        if (entry.isPresent()) {
+            return entry.get().isDefault();
+        }
+        return false;
+    }
+
+    public boolean cannotBeUpdatedDynamically() {
+        if (diff == null) {
+            return false;
+        } else return diff.anyReadOnly(kafkaVersion)
+                || !diff.unknownConfigs(kafkaVersion).isEmpty()
+                || diff.containsListenersChange();
+    }
+
+    /**
+     * @return Map object which is used for dynamic configuration of kafka broker
+     */
+    public Map<ConfigResource, Collection<AlterConfigOp>> getUpdatedConfig() {
+        return updated;
+    }
+
+    private boolean isIgnorableProperty(String key) {
+        return IGNORABLE_PROPERTIES.matcher(key).matches();
+    }
+
+    /**
+     * Computes diff between two maps. Entries in IGNORABLE_PROPERTIES are skipped
+     * @return KafkaConfiguration containing all entries which were changed from current in desired configuration
+     */
+    public KafkaConfiguration computeDiff() {
```

Comment:
I think this will cause lot of questions why did or didn't this roll. I think this needs to have debug level logging similar tot he statefulset diff.